server/hedley/modules/custom/hedley_reports/hedley_reports.module

```diff
@@ -5674,193 +5680,3072 @@ function hedley_reports_generate_completion_data_for_tuberculosis($participant,
+  $activity_expected = FALSE;
+
+  if ($encounter_type == 'nurse') {
+    // HIV education.
+    $hiv_test_measurement = $encounter_data['measurements_by_type']['prenatal_hiv_test'];
+    if (!empty($hiv_test_measurement)) {
+      $activity_expected = !empty($hiv_test_measurement->field_test_result[LANGUAGE_NONE][0]['value']);
+    }
+
+    // Partner presence education.
+    if (!$activity_expected) {
+      $partner_hiv_test_measurement = $encounter_data['measurements_by_type']['prenatal_partner_hiv_test'];
+      if (!empty($partner_hiv_test_measurement)) {
+        $activity_expected = $partner_hiv_test_measurement->field_test_execution_note[LANGUAGE_NONE][0]['value'] == 'not-present';
+      }
+    }
+
+    // Education per encounter diagnoses.
+    if (!$activity_expected) {
+      $triggering_diagnoses = array_merge(
+        hedley_reports_prenatal_diabetes_diagnoses(),
+        [
+          'heartburn',
+          'candidiasis',
+          'gonorrhea',
+          'trichomonas-or-bv',
+          'hiv-detectable-viral-load-initial',
+          'hiv-detectable-viral-load',
+        ]
+      );
+      $activity_expected = hedley_reports_prenatal_diagnosed_any_of($encounter_data, $triggering_diagnoses);
+    }
+
+    // Mental health education.
+    // Mental health survey is filled and none of mental health
+    // diagnoses requiring tratment was recorded.
+    $mental_health_measurement = $encounter_data['measurements_by_type']['prenatal_mental_health'];
+    if (!empty($mental_health_measurement)) {
+      $mental_health_diagnoses_requiring_treatment = hedley_reports_prenatal_mental_health_diagnoses_requiring_treatment();
+      $activity_expected = !hedley_reports_prenatal_diagnosed_any_of($encounter_data, $mental_health_diagnoses_requiring_treatment);
+    }
+  }
+  // Postpartum encounter.
+  else {
+    // Education per encounter diagnoses.
+    $triggering_diagnoses = [
+      'heartburn',
+      'candidiasis',
+      'gonorrhea',
+      'trichomonas-or-bv',
+      'postpartum-early-mastitis-engorgment',
+      'postpartum-mastitis',
+    ];
+    $activity_expected = hedley_reports_prenatal_diagnosed_any_of($encounter_data, $triggering_diagnoses);
+  }
+
+  // Nausea and Vomiting education.
+  $symptom_review_measurement = $encounter_data['measurements_by_type']['prenatal_symptom_review'];
+  if (!$activity_expected) {
+    // First, checking by current encounter.
+    // We need to make sure that nausea-and-vomiting symptom is recorded, and
+    // neither of follow-up questions (dizziness, low-urine-output
+    // and dark-urine) are answered positively.
+    // Then, we make sure that Nausea and Vomiting symptom was not recorded
+    // during previous encounters.
+    if (empty($symptom_review_measurement) || empty($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE])) {
+      $symptom_recorded = FALSE;
+      foreach ($symptom_review_measurement->field_prenatal_symptoms[LANGUAGE_NONE] as $symptom) {
```

Comment:
we do try to foreach over something empty, don't we?

Reply:
Yep. Good catch!  Fixed.